File: backend/src/main/java/com/sonnguyen/base/service/DeckService.java
package com.sonnguyen.base.service;

import com.sonnguyen.base.payload.request.CreateDeckRequest;
import com.sonnguyen.base.payload.response.DeckResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface DeckService {

    Page<DeckResponse> getMyDecks(Pageable pageable);

    Page<DeckResponse> getPublicDecks(Pageable pageable);

    DeckResponse getDeckById(String deckId);

    DeckResponse createDeck(CreateDeckRequest request);

    DeckResponse updateDeck(String deckId, CreateDeckRequest request);

    void deleteDeck(String deckId);
}

File: frontend/src/services/deck.service.js
import api from '@/config/api';

export const deckService = {
  /**
   * Get paginated list of the current user's decks
   */
  getMyDecks: async (page = 0, size = 20) => {
    const response = await api.get('/decks', { params: { page, size } });
    return response.data;
  },

  /**
   * Get paginated list of public decks
   */
  getPublicDecks: async (page = 0, size = 20) => {
    const response = await api.get('/decks/public', { params: { page, size } });
    return response.data;
  },

  /**
   * Get a single deck by ID
   */
  getDeckById: async (deckId) => {
    const response = await api.get(`/decks/${deckId}`);
    return response.data;
  },

  /**
   * Create a new deck
   * @param {{ title: string, description: string, isPublic: boolean }} data
   */
  createDeck: async (data) => {
    const response = await api.post('/decks', data);
    return response.data;
  },

  /**
   * Update an existing deck (owner only)
   */
  updateDeck: async (deckId, data) => {
    const response = await api.put(`/decks/${deckId}`, data);
    return response.data;
  },

  /**
   * Delete a deck (owner only)
   */
  deleteDeck: async (deckId) => {
    const response = await api.delete(`/decks/${deckId}`);
    return response.data;
  },
};
